src/pages/Admin/PermissionHandler.tsx

```diff
@@ -0,0 +1,33 @@
+import { ReactNode, useEffect, useState } from "react";
+import { matchPath, useLocation } from "react-router-dom";
+import { AdminResources } from "services/types";
+import { useGetWallet } from "contexts/WalletContext";
+import { adminRoutes, appRoutes } from "constants/routes";
+import { useAdminResources } from "./Guard";
+
+const ADMIN_ROUTE = `${appRoutes.admin}/:id/`;
+
+export default function PermissionHandler({
+  data,
+  children,
+}: {
+  data: AdminResources | undefined;
+  children: ReactNode;
+}) {
+  const { wallet } = useGetWallet();
+  const currPath = useLocation().pathname;
+  const [disableInputs, setDisableInputs] = useState(true);
+  const { whitelistedBeneficiaries, members } = useAdminResources<"charity">();
+  useEffect(() => {
+    if (wallet) {
+      const isMember = members.includes(wallet.address);
+      const isWhitelisted = whitelistedBeneficiaries.includes(wallet.address);
+      // allow deposits to any connected admin wallet
+      if (matchPath(`${ADMIN_ROUTE}${adminRoutes.deposits}`, currPath))
```

Comment:
IMO, it's better to set disabled on per template/form (more semantic use of `fieldset`) basis instead of disabling the whole admin page and adding some exceptions

> instead of disabling the whole admin page 

only makes sense if all form/templates have the same way of determining permissions